src/metabase/models/data_permissions/graph.clj

```diff
@@ -383,12 +312,34 @@
     (when new-db-perms
       (data-perms/set-database-permission! group-id db-id :perms/create-queries new-db-perms))))
 
+(defn- update-table-level-view-data-permissions!
```

Comment:
These are needed to support setting `legacy-no-self-service` at the table-level. It shouldn't be set in practice (and should be disallowed by the FE) but the BE should allow it for testing purposes.